client/src/components/EditAgent.js

```diff
@@ -0,0 +1,161 @@
+import { useParams, Link, useHistory } from 'react-router-dom'
+import { useEffect, useState, useContext } from 'react';
+import AuthContext from '../AuthContext';
+import Errors from './Errors';
+
+
+function EditAgent() {
+    const [firstName, setFirstName] = useState('');
+    const [middleName, setMiddleName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [dob, setDob] = useState('');
+    const [heightInInches, setHeightInInches] = useState('');
+    const [errors, setErrors] = useState([]);
+    
+    const auth = useContext(AuthContext);
+    
+    const { id } = useParams();
+    const history = useHistory(); 
+
+   
+
+    const firstNameOnChangeHandler = (event) => {
+        setFirstName(event.target.value);
+    }
+
+    const middleNameOnChangeHandler = (event) => {
+        setMiddleName(event.target.value);
+    }
+
+    const lastNameOnChangeHandler = (event) => {
+        setLastName(event.target.value);
+    }
+
+    const dobOnChangeHandler = (event) => {
+        setDob(event.target.value);
+    }
+
+    const heightInInchesOnChangeHandler = (event) => {
+        setHeightInInches(event.target.value);
+    }
+
+    useEffect(() => {
+        // GET http://localhost:8080/api/agent/1 HTTP/1.1
+
+        const init = {
+            headers: {
+                'Authorization': `Bearer ${auth.user.token}`
+            }
+        }
+
+        fetch(`http://localhost:8080/api/agent/${id}`, init)
+            // Response object
+            .then(response => {
+                if (response.status === 404) {
+                    return Promise.reject(`Received 404 Not Found for ToDo ID: ${id}`);
+                }
+                return response.json();
+            })
+            .then(data => {
+                setFirstName(data.firstName);
+                setMiddleName(data.middleName);
+                setLastName(data.lastName);
+                setDob(data.dob);
+                setHeightInInches(data.heightInInches);
```

Comment:
I wasn't able to figure this part out with the separate fields of the agent (firstName, middleName, etc) so I am very impressed with how this edit agent section turned out!!